lmfit/model.py

```diff
@@ -1618,7 +1618,10 @@ def dumps(self, **kws):
                      'nfree', 'nvarys', 'redchi', 'scale_covar', 'calc_covar',
                      'success', 'userargs', 'userkws', 'values', 'var_names',
                      'weights', 'user_options'):
-            val = getattr(self, attr)
```

Comment:
instead of the ```try```.... ```except``` clause, could we not just use ```val = getattr(self, attr, None)```. This will return ```None``` if the attribute does not exist (see [here](https://docs.python.org/3/library/functions.html#getattr)). That would likely set the attribute to ```None``` when loading the model again... I am not sure whether that would be an issue or not.

Reply:
I think it would be better to try to keep the results object the same after a save load cycle rather than switching a missing attribute to a None.

Reply:
I did try both solutions a bit more and agree with you, we should keep the objects the same and not set something to ```None``` as indeed would happen with my suggestion.